test/scylla_gdb/conftest.py

```diff
@@ -1,80 +1,93 @@
-# Copyright 2022-present ScyllaDB
+# Copyright 2025-present ScyllaDB
 #
 # SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+"""Conftest for Scylla GDB tests"""
 
-# This file configures pytest for all tests in this directory, and also
-# defines common test fixtures for all of them to use. A "fixture" is some
-# setup which an individual test requires to run; The fixture has setup code
-# and teardown code, and if multiple tests require the same fixture, it can
-# be set up only once - while still allowing the user to run individual tests
-# and automatically setting up the fixtures they need.
-
+import subprocess
 import pytest
-import os
-import sys
 
-try:
-    import gdb as gdb_library
-except:
-    print('This test must be run inside gdb. Run ./run instead.')
-    exit(1)
+from test.cqlpy import run
+
+
+@pytest.fixture(scope="package")
+def scylla_pid():
+    """
+    Runs Scylla and waits until it can respond to CQL queries.
+    Skips the test if the Scylla executable lacks debugging symbols.
+    """
+    if (
+        ".debug_info"
+        not in subprocess.run(
+            ["objdump", "-h", run.find_scylla()], capture_output=True, text=True
+        ).stdout
+    ):
+        pytest.skip(
+            "Scylla executable was compiled without debugging symbols (-g). "
+            "Set the SCYLLA environment variable to a debug build to run GDB tests."
+        )
+
+    pid = run.run_with_temporary_dir(run.run_scylla_cmd)
+    ip = run.pid_to_ip(pid)
+    run.wait_for_services(pid, [lambda: run.check_cql(ip)])
+    return pid
+
+
+@pytest.fixture(scope="package")
+def gdb_config(scylla_pid):
+    """
+    Prepares the GDB configuration for attaching to the Scylla process.
+    Imports scylla-gdb.py to the GDB
+    """
+    args = [
+        "gdb",
+        "-batch",
+        "-n",
+        "-se",
+        run.scylla,
+        "-p",
+        str(scylla_pid),
+        "-ex",
+        "set auto-load safe-path /",
+        "-ex",
+        "set python print-stack full",
+        "-x",
+        "../../scylla-gdb.py",
+    ]
+
+    return args
+
+
+@pytest.fixture(scope="package")
```

Comment:
why package? That is a weird scope

Reply:
Why weird? All fixtures relying on this fixture are in the same package, no need to extend it to `session`

Reply:
it is unusual I would say, but I guess it is fine here

Reply:
agree, that is not usual, but fits here